libs/langchain/langchain/llms/fireworks.py

```diff
@@ -179,6 +228,74 @@ async def _completion_with_retry(**kwargs: Any) -> Any:
     return await _completion_with_retry(**kwargs)
 
 
+def completion_with_retry_batching(
```

Comment:
we handle retries via -- fireworks should not contain any logic for retries

https://api.python.langchain.com/en/latest/schema.runnable/langchain.schema.runnable.retry.RunnableRetry.html

Reply:
fireworks.client has own error handling wrapper
e.g. in `_create_retry_decorator`
```
fireworks.client.error.RateLimitError,
fireworks.client.error.ServiceUnavailableError,
```

Can we keep it to use our fireworks library and use the custom error handling?

Reply:
RunnableRetry allows using custom exceptions; 

Once Fireworks inherits from BaseLLM, one can do this:

```python
llm = Fireworks(..).with_retry() # Will automatically retry any exception 3 times
```

```python
retryable_exceptions = (fireworks.client.error.RateLimitError, fireworks.client.error.ServiceUnavailableError)
llm = Fireworks(..).with_retry(retry_if_exception_type=retryable_exceptions) # Will automatically retry any exception 3 times
```

Would either of these options work?

Reply:
Got it. But then users have to handle retries themselves. We handled retries from the beginning. Is there any way we can still handle retries on our side to avoid breaking users' workflow?

Reply:
Following the comment in the slack, we keep out retry mechanism but allow users to opt out with `use_retry` argument.